packages/react-components/react-motion/library/src/types.ts

```diff
@@ -1,4 +1,15 @@
-export type AtomMotion = { keyframes: Keyframe[] } & KeyframeEffectOptions;
+export type AtomMotion = { keyframes: Keyframe[] } & KeyframeEffectOptions & {
+    /**
+     * Allows to specify a reduced motion version of the animation. If provided, the settings will be used when the
+     * user has enabled the reduced motion setting in the operating system. By default, the duration of the animation is
+     * set to 1ms.
+     *
+     * Note, if keyframes are provided, they will be used instead of the regular keyframes.
+     *
+     * @see https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion
+     */
+    reducedMotion?: { keyframes?: Keyframe[] } & KeyframeEffectOptions;
```

Comment:
Just throwing out another idea, instead of making the reduced motion options a recursion of the parent motion declaration we use a second argument in the factory and enforce the SRP for motion declarations?

```ts
createPresentionComponent(atom, reducedMotionAtom?)
```

This way we'll avoid TS recursion, and decouple the original motion and its reduced motion alternative. From what I understand even the current `reducedMotion` property completely overrides all other options anyway

Reply:
It's not recursion though; it's just leaf reuse. There is no atom inside an atom, nor keyframe inside a keyframe.

Reply:
@ling1726 please check "rejected options" in the PR's description, the order issue will be the same here.